pandas/core/nanops.py

```diff
@@ -587,8 +584,12 @@ def get_median(x):
         return np.nanmedian(x[mask])
 
     values, mask, dtype, dtype_max, _ = _get_values(values, skipna, mask=mask)
-    if not is_float_dtype(values):
-        values = values.astype("f8")
+    if not is_float_dtype(values.dtype):
+        try:
```

Comment:
can you use _ensure_numeric?

Reply:
will give it a shot.  that would change behavior if we have complex dtype, which i dont think median should support

Reply:
changing this doesnt appear to break any tests.  will change in a follow-up (dont want to push if i can avoid it since the CI is failing for unrelated reasons)